db/seeds.d/020-provisioning_templates_list.rb

```diff
@@ -94,13 +109,17 @@ def seeded_templates
     end
 
     def kinds
-      kinds = {}
-      TemplateKind.default_template_labels.keys.map(&:to_sym).each do |type|
-        kinds[type] = TemplateKind.unscoped.find_by_name(type)
-        kinds[type] ||= TemplateKind.unscoped.create(:name => type)
-        raise "Unable to create template kind: #{format_errors kinds[type]}" if kinds[type].nil? || kinds[type].errors.any?
+      @kinds ||= begin
+        kinds = {}
+        TemplateKind.default_template_labels.keys.map(&:to_sym).each do |type|
+          return kinds[type] if kinds.key?(type)
```

Comment:
Can you explain this return? I believe that it should never trigger as you iterate through unique keys. Let's remove it?

Reply:
This return was mentioned with the thought of running db:seed post the data is already in table (i.e. not the first time), so that if key in that hash already exists, it would not execute any further steps. But, I agree, we can remove it.

Reply:
Also, for this seed, I was planning to write tests in [GitHub](https://github.com/theforeman/foreman/blob/develop/test/unit/tasks/seeds_test.rb) .
Is that the correct? Any suggestions / ideas on test ?

Reply:
I don't think this feature requires a test - description missing is non-fatal.